asistente_ladm_col/logic/quality_rules/qr_administrative_source_with_duplicate_records.py

```diff
@@ -0,0 +1,107 @@
+        email                : contacto@ceicol.com
+
+ ***************************************************************************/
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License v3.0 as          *
+ *   published by the Free Software Foundation.                            *
+ *                                                                         *
+ ***************************************************************************/
+"""
+from qgis.PyQt.QtCore import (QCoreApplication,
+                              QSettings)
+
+from asistente_ladm_col.config.enums import EnumQualityRuleResult
+from asistente_ladm_col.config.keys.common import QUALITY_RULE_LADM_COL_LAYERS
+from asistente_ladm_col.config.layer_config import LADMNames
+from asistente_ladm_col.config.quality_rule_config import (QR_IGACR4022,
+                                                           QRE_IGACR4022E01)
+from asistente_ladm_col.core.quality_rules.abstract_logic_quality_rule import AbstractLogicQualityRule
+from asistente_ladm_col.core.quality_rules.quality_rule_execution_result import QualityRuleExecutionResult
+from asistente_ladm_col.logic.ladm_col.ladm_data import LADMData
+
+
+class QRAdministrativeSourceWithDuplicateRecords(AbstractLogicQualityRule):
+    """
+    Check that administrative source don't have duplicate records
+    """
+    _ERROR_01 = QRE_IGACR4022E01  # Administrative source with duplicate records
+
+    def __init__(self):
+        AbstractLogicQualityRule.__init__(self)
+
+        self._id = QR_IGACR4022
+        self._name = "Fuente administrativa no debe tener registros duplicados"
+        self._tags = ["igac", "instituto geográfico agustín codazzi", "lógica",
+                      "negocio", "fuente administrativa", "duplicado"]
+        self._models = [LADMNames.SURVEY_MODEL_KEY]
+
+        self._errors = {self._ERROR_01: "Fuente Administrativa no debe tener registros repetidos"}
+
+        # Optional. Only useful for display purposes.
+        self._field_mapping = dict()  # E.g., {'id_objetos': 'ids_punto_lindero', 'valores': 'conteo'}
+
+    def layers_config(self, names):
+        return {QUALITY_RULE_LADM_COL_LAYERS: [names.LC_ADMINISTRATIVE_SOURCE_T]}
+
+    def _validate(self, db, db_qr, layer_dict, tolerance, **kwargs):
+        self.progress_changed.emit(5)
+        ladm_queries = self._get_ladm_queries(db.engine)
+
+        pre_res, pre_obj = self._check_prerrequisite_layers(layer_dict)
+        if not pre_res:
+            return pre_obj
+
+        error_state = None
+
+        # Check administrative source with duplicate records
+        table = db.names.LC_ADMINISTRATIVE_SOURCE_T
+        fields = [db.names.LC_ADMINISTRATIVE_SOURCE_T_EMITTING_ENTITY_F,
```

Comment:
I think we can use this function to get the minimum structure for each table `fields = LayerConfig.get_logic_consistency_tables(db.names).get(db.names.LC_ADMINISTRATIVE_SOURCE_T)`

Reply:
you can get this method in asistente_ladm_col.config.layer_config -> LayerConfig

Reply:
I think it is a better strategy to eliminate the LayerConfig.get_logic_consistency_tables() method considering that now with the code structure we can have a class for each quality rule that abstracts the logic. Thinking in the future we can have other models for which we want to validate that there are no duplicate records and we cannot use the LayerConfig.get_logic_consistency_tables() method because the tables of the survey model will not exist.

Reply:
Ok, you're right, I'll replicate this in my PR.